### PR TITLE
aws-workspaces: init at 3.1.3.925

### PR DESCRIPTION
**Pull request by mausch**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114162` at 42e4ff68 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/254cdcb2e5fc2f2357969d2221207866">aws-workspaces</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/networking/remote/aws-workspaces/default.nix:43:3:
```
   |
43 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-uri-literals.md">no-uri-literals</a>

URI literals are deprecated.
Near pkgs/applications/networking/remote/aws-workspaces/default.nix:55:16:
```
   |
55 |     homepage = https://clients.amazonworkspaces.com/;
   |                ^
```
</li>
  </ul>
</details>

File: pkgs/applications/networking/remote/aws-workspaces/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+stdenv.mkDerivation rec {
+  pname = "aws-workspaces";
+  version = "3.1.0.813";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+  ];
+
+  dontStrip = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/aws-workspaces/default.nix:15]:
```suggestion
```
Is there a reason why we should not strip?

File: pkgs/applications/networking/remote/aws-workspaces/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    gdk-pixbuf
+    libsoup
+    glib-networking
+  ];
+
+  src = fetchurl {
+    # ref https://d3nt0h4h6pmmc4.cloudfront.net/ubuntu/dists/bionic/main/binary-amd64/Packages
+    url = "https://d3nt0h4h6pmmc4.cloudfront.net/ubuntu/dists/bionic/main/binary-amd64/workspacesclient_${version}_amd64.deb";
+    sha256 = "39ef7117f835fd58e578a4a374f6c2bad18588a36a91d94f3b41c49e947260ff";
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/aws-workspaces/default.nix:37]:
```suggestion
```

File: pkgs/applications/networking/remote/aws-workspaces/default.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
+{ stdenv, lib
+, makeWrapper, dpkg, fetchurl, autoPatchelfHook
+, curl, kerberos, lttng-ust, libpulseaudio, gtk3, openssl_1_1, icu, webkitgtk, librsvg, gdk-pixbuf, libsoup, glib-networking
+}:
+
+stdenv.mkDerivation rec {
+  pname = "aws-workspaces";
+  version = "3.1.0.813";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/aws-workspaces/default.nix:8]:
```suggestion
  version = "3.1.0.813";

  src = fetchurl {
    # ref https://d3nt0h4h6pmmc4.cloudfront.net/ubuntu/dists/bionic/main/binary-amd64/Packages
    url = "https://d3nt0h4h6pmmc4.cloudfront.net/ubuntu/dists/bionic/main/binary-amd64/workspacesclient_${version}_amd64.deb";
    sha256 = "39ef7117f835fd58e578a4a374f6c2bad18588a36a91d94f3b41c49e947260ff";
  };
```

File: pkgs/applications/networking/remote/aws-workspaces/default.nix (hunk 1 above)
Comment (mausch) [reply pkgs/applications/networking/remote/aws-workspaces/default.nix:15]: Yep, a very good reason actually: it crashes at startup when I do it :smile: 
`Failed to create CoreCLR, HRESULT: 0x80004005`

File: pkgs/applications/networking/remote/aws-workspaces/default.nix (hunk 2 above)
Comment (mausch) [reply pkgs/applications/networking/remote/aws-workspaces/default.nix]: Applied in e035973faabfba9299563525986e829ef5753391

File: pkgs/applications/networking/remote/aws-workspaces/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/networking/remote/aws-workspaces/default.nix:21]: Can you add a comment with that information?

File: pkgs/applications/networking/remote/aws-workspaces/default.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    mv $out/opt/workspacesclient/* $out/bin
+
+    wrapProgram $out/bin/workspacesclient \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath buildInputs}" \
+      --set GDK_PIXBUF_MODULE_FILE "${librsvg.out}/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache" \
+      --set GIO_EXTRA_MODULES "${glib-networking.out}/lib/gio/modules"
+  '';
+
+  meta = with lib; {
+    description = "Client for Amazon WorkSpaces, a managed, secure Desktop-as-a-Service (DaaS) solution.";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/aws-workspaces/default.nix:54]:
```suggestion
    description = "Client for Amazon WorkSpaces, a managed, secure Desktop-as-a-Service (DaaS) solution";
```

File: pkgs/applications/networking/remote/aws-workspaces/default.nix (hunk 1 above)
Comment (mausch) [reply pkgs/applications/networking/remote/aws-workspaces/default.nix]: Done in 4b1675c784bccf503d8623eccb4b47050dc37ba7

File: pkgs/applications/networking/remote/aws-workspaces/default.nix (hunk 4 above)
Comment (mausch) [reply pkgs/applications/networking/remote/aws-workspaces/default.nix]: Applied in 4b1675c784bccf503d8623eccb4b47050dc37ba7

**Comment by SuperSandro2000** [comment]
```
builder for '/nix/store/pzbc2k6gj2475lvznmish42nbbisyaz6-workspacesclient_3.1.0.813_amd64.deb.drv' failed with exit code 1; last 7 log lines:

  trying https://d3nt0h4h6pmmc4.cloudfront.net/ubuntu/dists/bionic/main/binary-amd64/workspacesclient_3.1.0.813_amd64.deb
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  curl: (22) The requested URL returned error: 403
  error: cannot download workspacesclient_3.1.0.813_amd64.deb from any mirror
cannot build derivation '/nix/store/j207ahc2fg1sj5kbgadi7iw3sy94lzlk-aws-workspaces-3.1.0.813.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/chrfni1bidmw8h4zajr1fimy1jmj0bkb-review-shell.drv': 1 dependencies couldn't be built
```

**Comment by mausch** [comment]
That version is now no longer available from AWS. Updated the version in b1c93c3942799f0f9caec562b2aee548fa1a25b8 .

File: pkgs/applications/networking/remote/aws-workspaces/default.nix (hunk 1 above)
Comment (mausch) [reply pkgs/applications/networking/remote/aws-workspaces/default.nix]: BTW this is the case with most (all?) .NET applications. Haven't looked into why.

**Comment by SuperSandro2000** [comment]
> That version is now no longer available from AWS. Updated the version in [b1c93c3](https://github.com/NixOS/nixpkgs/commit/b1c93c3942799f0f9caec562b2aee548fa1a25b8) .

Can you upload the file to web.archive.org and add it as a second link just incase this happens again?

**Comment by mausch** [comment]
> > That version is now no longer available from AWS. Updated the version in [b1c93c3](https://github.com/NixOS/nixpkgs/commit/b1c93c3942799f0f9caec562b2aee548fa1a25b8) .
> 
> Can you upload the file to web.archive.org and add it as a second link just incase this happens again?

Done in 128a37174f8091d7a7f00ecb8daf91dd02f9c64c

**Pull request merged**